ouroboros-network/test/Test/Mux.hs

```diff
@@ -67,7 +68,7 @@ startMuxSTM mpds wq rq mtu = do
     let ctx = MuxSTMCtx wq rq mtu
     jobs <- Mx.muxJobs mpds (writeMux ctx) (readMux ctx) (sduSizeMux ctx)
     aids <- mapM async jobs
-    fork (watcher aids)
+    void $ fork (watcher aids)
```

Comment:
Or we have `async` now. Not necessary to do in this PR. We can do some general fork/async clean up separately.

Reply:
I agree that with the async changes the cleanup can be shared between Socket, Pipes, and MuxSTM. But that is a separate PR.